src/restore.c

```diff
@@ -1168,6 +1169,13 @@ check_index(aerospike *as, index_param *index, uint32_t timeout)
 		}
 	}
 
+	if (!index->ctx && index->ctx != index2.ctx) {
```

Comment:
condition "!index->ctx" needed to be refined. 
get rid of the first check
check str<>str not pointer to pointer